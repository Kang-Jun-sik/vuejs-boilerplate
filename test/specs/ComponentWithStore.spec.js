import {
  shallow,
} from 'vue-test-utils';
import {
  createRenderer,
} from 'vue-server-renderer';

import store from '@/store';
import ComponentWithStore from '@/components/ComponentWithStore';

describe('Component.vue', () => {
  it('renders the state from the store', () => {
    const { vm } = shallow(ComponentWithStore, {
      store,
    });
    expect(vm.count).toBe(0);
  });

  it('updates the store by calling the increment method', () => {
    const { vm } = shallow(ComponentWithStore, {
      store,
    });
    vm.increment();
    expect(vm.count).toBe(1);
  });

  it('matches snapshot', () => {
    store.state.count = 0; // reset state
    const renderer = createRenderer();
    const { vm } = shallow(ComponentWithStore, {
      store,
    });
    renderer.renderToString(vm, (err, str) => {
      if (err) throw new Error(err);
      expect(str).toMatchSnapshot();
    });
  });
});
