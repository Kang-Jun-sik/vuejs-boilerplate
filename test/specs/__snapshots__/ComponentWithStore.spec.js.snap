// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component.vue matches snapshot 1`] = `
<div>
    0
    <button>+1</button></div>
`;
